controllers/clustertopology_controller_compute_test.go

```diff
@@ -17,15 +17,257 @@ limitations under the License.
 package controllers
 
 import (
+	"context"
 	"testing"
 
+	"github.com/google/go-cmp/cmp"
 	. "github.com/onsi/gomega"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
+func TestGetClass(t *testing.T) {
```

Comment:
Q: What about having a test that verifies getTemplate and ensures version conversion once for all, so this test can just focus on checking that we get all the objects we need (and we fail whenever one is missing)?

Reply:
Sounds like a good idea, especially after we stopped patching the ref in ControlPlaneTemplate